Filter elections endpoint by org identifier

michaeljcollinsuk opened the pull request:
- Reorganise ElectionViewset filterset_fields to ElectionFilter class
- Add organisation_identifier filter
- Closes #1151

symroe reviewed (approved):
:+1:

chris48s commented:
Organisations in Every Election have a combined key `(organisation_type, official_identifier, start_date)`.
See https://github.com/DemocracyClub/EveryElection/issues/182

This is why the API URL for Aberdeen (for example) has to be
https://elections.democracyclub.org.uk/api/organisations/local-authority/ABE/1996-04-01/
not
https://elections.democracyclub.org.uk/api/organisations/ABE

This filter is going to give you issues in situations where `official_identifier` in non-unique (which is rare but there is a non-zero number of cases and more could happen) because you'll get >1 organisations worth of elections merged together (which I don't think is the intent of the feature).

My suggestion for how to do avoid this problem in a nice way would be:

Instead of going from

https://elections.democracyclub.org.uk/api/elections?organisation_identifier=ABE
to
https://elections.democracyclub.org.uk/api/elections?organisation_type=local-authority&official_identifier=ABE&start_date=1996-04-01

to solve the issue, you make the endpoint

https://elections.democracyclub.org.uk/api/organisations/local-authority/ABE/1996-04-01/elections

michaeljcollinsuk commented:
Thanks for the additional context @chris48s 

@symroe I have updated to allow add additional filters to the Elections endpoint, and implemented the `/elections` endpoint on the OrganisationViewSet
